source/core/coverage/ut_coverage.pkb

```diff
@@ -65,68 +65,104 @@ create or replace package body ut_coverage is
       g_skipped_objects(g_skipped_objects.last) := a_object;
   end;
 
-  function get_coverage_data(a_coverage_id integer := null) return t_coverage is
+  function get_coverage_data return t_coverage is
 
     type t_coverage_row is record(
       name          varchar2(500),
       line_number   integer,
       total_occur   number(38,0)
     );
     type tt_coverage_rows is table of t_coverage_row;
-
-    l_data             ut_coverage_rows;
+    l_line_calls       ut_coverage_helper.unit_line_calls;
     l_result           t_coverage;
     l_new_unit         t_unit_coverage;
     l_skipped_objects  ut_object_names := ut_object_names();
+
+    type t_source_lines is table of binary_integer;
+    l_source_lines     t_source_lines;
+    line_no            binary_integer;
   begin
 
     if not ut_coverage_helper.is_develop_mode() then
       l_skipped_objects := ut_utils.get_utplsql_objects_list() multiset union set(g_skipped_objects);
     end if;
 
-    -- TODO - add inclusive and exclusive filtering
-    select ut_coverage_row(
-             lower(s.owner||'.'||s.name),
-             s.owner, s.name, s.type, s.line,
-           --filtering out false - negatives reported by profiler (zero executions while it should be ignored).
-           case when
-              regexp_instr(
-                s.text,
-                '^\s*(((not)?\s*(overriding|final|instantiable)\s*)*(constructor|member)?\s*(procedure|function)|begin|end\s*;)', 1, 1, 0, 'i'
-              ) = 0 then u.total_occur
-           end
-          )
-      bulk collect into l_data
+    --prepare global temp table with sources
+    delete from ut_coverage_sources_tmp;
+
+    insert into ut_coverage_sources_tmp(owner,name,line,text)
+    select s.owner,s.name,s.line,s.text
       from all_source s
-      --need to use owner privileges here
-      left join table( ut_coverage_helper.get_raw_coverage_data() ) u
-        on s.owner = u.owner
-       and s.name  = u.name
-       and s.type  = u.type
-       and s.line  = nvl(u.line_number, s.line)
      where s.type not in ('PACKAGE', 'TYPE')
        and s.owner in (select t.column_value from table(g_schema_names) t)
```

Comment:
I recommend to put /*+dynamic_sampling(t 2)*/ for optimized to start with the size of collection, or we can set /*+cardinality(5)*/ which is not as good to mark that the list of schemas is small